### DISCUSSION TITLE
Deploy beta to TestFlight using env.staging

### DISCUSSION THREAD

**Discussion by felipepalazzo** [goatandsheep/react-native-dotenv#299] [Q&A] [answered]
I want to deploy a Beta version to TestFlight using the env variables in `.env.staging`. I have the following script in my packge.json: 

```
"start:staging": "APP_ENV=staging react-native start",
```

If I run `yarn start:staging` in my terminal and, after that, `npx react-native run-ios`, it opens the ios simulator and it's using the staging variables as expected. 

Going to Xcode and doing the process to deploy to TestFlight, i.e. Product > Archive > and so on, it always builds with the .env file.

Is there a way to do that?

**Comment by felipepalazzo** [comment]
Just complementing my question above, I understand that Release == production and Debug == development, so I'm assuming that's no way to build in Xcode using `env.staging` (or anything other than env.development / env.production)

**Reply by goatandsheep** [reply]
no there's a couple options. The option I recommend is using schemes (https://shockoe.com/ideas/development/how-to-setup-configurations-and-schemes-in-xcode/). Basically you can specify your NODE_ENV in your build process.

If that doesn't work for whatever reason, you can do the same thing but add APP_ENV (there's some docs on this in teh readme)

**Comment by goatandsheep** [comment, marked as answer]
there's a couple options. The option I recommend is using schemes (https://shockoe.com/ideas/development/how-to-setup-configurations-and-schemes-in-xcode/). Basically you can specify your NODE_ENV in your build process.

If that doesn't work for whatever reason, you can do the same thing but add APP_ENV (there's some docs on this in teh readme)

**Comment by felipejh** [comment]
@felipepalazzo how did you solve your problem? I've been trying it but with no success.
I've configurate the schemes. Did you created environment variables?

**Reply by goatandsheep** [reply]
Can you attach a test environment? I'm unable to recreate it not working.
